auth-web/src/components/pay/CompletePaymentDetails.vue

```diff
@@ -142,41 +186,106 @@ export default defineComponent({
+        return true
+      } else {
+        state.errorMessage = 'Bank information validation failed'
+        if (verifyPad?.message?.length) {
+          let msgList = ''
+          verifyPad.message.forEach((msg) => {
+            msgList += `${msg}`
+          })
+          state.errorMessage = `${msgList}`
+        }
+        return false
+      }
+    }
+
+    async function getCreateRequestBody () {
+      let isValid = false
+      let createRequestBody: CreateRequestBody
+
+      if (props.changePaymentType === PaymentTypes.BCOL) {
+        isValid = !!(state.bcolInfo.userId && state.bcolInfo.password)
+        if (!isValid) {
+          state.errorMessage = 'Missing User ID and Password for BC Online.'
+          return
+        }
+        createRequestBody = {
+          paymentInfo: {
+            paymentMethod: PaymentTypes.BCOL
+          },
+          bcOnlineCredential: state.bcolInfo
+        }
+      } else if (props.changePaymentType === PaymentTypes.PAD && state.padValid) {
+        isValid = await verifyPAD()
+        createRequestBody = {
+          paymentInfo: {
+            paymentMethod: PaymentTypes.PAD,
+            bankTransitNumber: state.padInfo.bankTransitNumber,
+            bankInstitutionNumber: state.padInfo.bankInstitutionNumber,
+            bankAccountNumber: state.padInfo.bankAccountNumber
+          }
+        }
+      }
+      return { isValid, createRequestBody }
+    }
+
+    function getPADInfo (padInfoValue: PADInfo) {
+      state.padInfo = padInfoValue
+    }
+
+    async function complete () {
+      state.isLoading = true
+      state.checkErrors = true
+      state.isInitialTOSAccepted = true
       state.errorMessage = ''
       if (props.changePaymentType === PaymentTypes.BCOL) {
         const isValid = !!(state.bcolInfo.userId && state.bcolInfo.password)
         if (!isValid) {
           state.errorMessage = 'Missing User ID and Password for BC Online.'
+          state.isLoading = false
           return
         }
       }
-      const createRequestBody = {
-        paymentInfo: {
-          paymentMethod: PaymentTypes.BCOL
-        },
-        bcOnlineCredential: state.bcolInfo
-      }
-      try {
-        await orgStore.updateOrg(createRequestBody)
-        goToPaymentOptions()
-      } catch (error) {
-        switch (error.response.status) {
-          case 400:
-          case 409:
-            state.errorMessage = error.response.data.message
-            break
-          default:
-            state.errorMessage = 'An error occurred while attempting to create your account.'
+      const { isValid, createRequestBody } = await getCreateRequestBody()
+
+      if (isValid) {
+        try {
+          await orgStore.updateOrg(createRequestBody)
+          orgStore.setCurrentOrganizationPaymentType(props.changePaymentType)
+          state.isPaymentChanged = true
+          goToPaymentOptions()
+        } catch (error) {
+          switch (error.response.status) {
```

Comment:
Should probably do something like this for the two requests above, I've see banking validation fail before